mediatek/filogic: mtk-eth-soc add initial support for jumbo frames on mt7988

rmandrad opened the pull request:
Device Compatibility Check:

    Introduced is_mt7988_eth to determine compatibility with the mediatek,mt7988-eth hardware.
    Includes logging messages for debugging compatibility during initialization.

Enhanced MTU Configuration:

    Adjusted eth->netdev[id]->max_mtu to support jumbo frames if is_mt7988_eth is true.
    Default MTU remains the same for other device types.

Dynamic RX Frame Size Handling:

    Modified logic in mtk_mac_config to accommodate jumbo frames and extended frame size configurations:
        For MTK_GDM_TYPE, jumbo frames are enabled when is_mt7988_eth is true.
        Added logic to handle extended frame sizes for MTK_XGDM_TYPE using the new MTK_XMAC_RX_CFG2.

Initialization Updates:

    Extended device initialization to verify compatibility and set is_mt7988_eth accordingly.

Enhanced Register Configuration:

    Modified MAC control register (MAC_MCR) handling:
        Configures RX frame sizes based on the new maximum frame size (MTK_MAX_RX_LENGTH_9K) if is_mt7988_eth is enabled.
    Configures the MTK_XMAC_RX_CFG2 register for MTK_XGDM_TYPE devices.

tested on: filogic/bananapi4

Signed-off-by: Rudy Andram <rmandrad@gmail.com>

rmandrad committed:
mediatek: mtk-eth-soc add initial support for jumbo frames on mt7988

Device Compatibility Check:

    Introduced is_mt7988_eth to determine compatibility with the mediatek,mt7988-eth hardware.
    Includes logging messages for debugging compatibility during initialization.

Enhanced MTU Configuration:

    Adjusted eth->netdev[id]->max_mtu to support jumbo frames if is_mt7988_eth is true.
    Default MTU remains the same for other device types.

Dynamic RX Frame Size Handling:

    Modified logic in mtk_mac_config to accommodate jumbo frames and extended frame size configurations:
        For MTK_GDM_TYPE, jumbo frames are enabled when is_mt7988_eth is true.
        Added logic to handle extended frame sizes for MTK_XGDM_TYPE using the new MTK_XMAC_RX_CFG2.

Initialization Updates:

    Extended device initialization to verify compatibility and set is_mt7988_eth accordingly.

Enhanced Register Configuration:

    Modified MAC control register (MAC_MCR) handling:
        Configures RX frame sizes based on the new maximum frame size (MTK_MAX_RX_LENGTH_9K) if is_mt7988_eth is enabled.
    Configures the MTK_XMAC_RX_CFG2 register for MTK_XGDM_TYPE devices.

tested on: filogic/bananapi4

Signed-off-by: Rudy Andram <rmandrad@gmail.com>

namiltd commented on target/linux/mediatek/patches-6.6/750-mtk-eth-add-jumbo-frame-support-mt7998.patch:
```diff
@@ -0,0 +1,157 @@
+ 
++	// Check for compatibility with "mediatek,mt7988-eth"
++	if (!of_device_is_compatible(pdev->dev.of_node, "mediatek,mt7988-eth")) {
++		dev_err(&pdev->dev, "Device not compatible with mediatek,mt7988-eth\n");
```
Unnecessary information on other chipsets.

rmandrad replied:
removed

namiltd commented on target/linux/mediatek/patches-6.6/750-mtk-eth-add-jumbo-frame-support-mt7998.patch:
```diff
@@ -0,0 +1,158 @@
++	if (!of_device_is_compatible(pdev->dev.of_node, "mediatek,mt7988-eth")) {;
++		is_mt7988_eth = false;
++	} else {
++		dev_info(&pdev->dev, "Device compatible with mediatek,mt7988-eth - Jumbo frames capable\n");
```
+          dev_info(&pdev->dev, "Device compatible with mediatek,mt7988-eth - Jumbo frames 9216 capable\n");

rmandrad committed:
fixup! mediatek: mtk-eth-soc add initial support for jumbo frames on mt7988

rmandrad committed:
amend! mediatek/filogic: mtk-eth-soc add initial support for jumbo frames on mt7988

mediatek: mtk-eth-soc add initial support for jumbo frames on mt7988

Device Compatibility Check:

    Introduced is_mt7988_eth to determine compatibility with the mediatek,mt7988-eth hardware.
    Includes logging messages for debugging compatibility during initialization.

Enhanced MTU Configuration:

    Adjusted eth->netdev[id]->max_mtu to support jumbo frames if is_mt7988_eth is true.
    Default MTU remains the same for other device types.

Dynamic RX Frame Size Handling:

    Modified logic in mtk_mac_config to accommodate jumbo frames and extended frame size configurations:
        For MTK_GDM_TYPE, jumbo frames are enabled when is_mt7988_eth is true.
        Added logic to handle extended frame sizes for MTK_XGDM_TYPE using the new MTK_XMAC_RX_CFG2.

Initialization Updates:

    Extended device initialization to verify compatibility and set is_mt7988_eth accordingly.

Enhanced Register Configuration:

    Modified MAC control register (MAC_MCR) handling:
        Configures RX frame sizes based on the new maximum frame size (MTK_MAX_RX_LENGTH_9K) if is_mt7988_eth is enabled.
    Configures the MTK_XMAC_RX_CFG2 register for MTK_XGDM_TYPE devices.

tested on: filogic/bananapi4

Signed-off-by: Rudy Andram <rmandrad@gmail.com>
